core/src/main/java/org/mule/runtime/core/internal/streaming/object/AbstractCursorIteratorProvider.java

```diff
@@ -23,22 +28,45 @@
 
   protected final Iterator stream;
   private final AtomicBoolean closed = new AtomicBoolean(false);
+  private final ComponentLocation originatingLocation;
+  private Exception closerResponsible;
+
+  private static final boolean TRACK_CURSOR_PROVIDER_CLOSE = getBoolean(TRACK_CURSOR_PROVIDER_CLOSE_PROPERTY);
```

Comment:
same as before. The SP should be consumed in one unique spot and then propagated through constructors